diff --git a/src/redis_ipc.c b/src/redis_ipc.c
index 84d7c2a..cfb8574 100644
--- a/src/redis_ipc.c
+++ b/src/redis_ipc.c
@@ -27,7 +27,7 @@ static __thread struct redis_ipc_per_thread *redis_ipc_info = NULL;
 
 // gettid() is missing a libc wrapper for some reason
 // (manpage even mentions it)
-static pid_t gettid()
+pid_t gettid()
 {
   return syscall(SYS_gettid);
 }
@@ -381,7 +381,6 @@ static json_object * redis_pop(const char *queue_path, unsigned int timeout)
 
     // parse popped entry back into json object
     entry = json_tokener_parse(json_text);
-    if (is_error(entry)) entry = NULL;
 
 redis_pop_finish:
     if (reply != NULL)
@@ -1252,7 +1251,6 @@ json_object * redis_ipc_get_message_blocking(void)
 
     // parse message back into json object
     message = json_tokener_parse(message_str);
-    if (is_error(message)) message = NULL;
 
 redis_get_channel_message_finish:
     if (reply != NULL)
diff --git a/test/multithread_test.c b/test/multithread_test.c
index 7ec0b62..3eb1e0a 100644
--- a/test/multithread_test.c
+++ b/test/multithread_test.c
@@ -7,7 +7,7 @@
 
 // gettid() is missing a libc wrapper for some reason
 // (manpage even mentions it)
-static pid_t gettid()
+pid_t gettid()
 {
   return syscall(SYS_gettid);
 }
